client/my-sites/add-ons/hooks/use-add-ons.ts

```diff
@@ -0,0 +1,55 @@
+import { WPCOM_FEATURES_NO_ADVERTS } from '@automattic/calypso-products';
+import { useSelector } from 'react-redux';
+import { getProductBySlug, getProductName } from 'calypso/state/products-list/selectors';
+import noAdsIcon from '../icons/no-ads';
+
+export interface AddOnMeta {
+	slug: string;
+	name: string;
+	description: string;
+	highlight?: boolean;
+	icon: JSX.Element;
+}
+
+// these are pulled from API in the hook below
+const addOnsActive = [
+	{
+		slug: WPCOM_FEATURES_NO_ADVERTS,
+		highlight: false,
+		nameOverride: 'Remove Ads',
```

Comment:
We'll want to pass in translated strings, for which, I'm not sure we can process outside of a component.

Reply:
Moved this structure inside the hook. 

The other problem I'm not too sure of is using the API metadata (name, description, currency, term, etc.). Not sure how translations go with that:

- it looks like not everything is translated (no-ads description doesn't bring back Spanish)
- some state refresh may be needed when switching locales, cause it lingers to previous on my end

May need to investigate a bit